src/utils/authMiddlewares.js

```diff
@@ -210,16 +212,26 @@ const authenticationMiddlewareSocketIO = async (authHeader) => {
  * and remove authByExperiment
  */
 const authorize = async (userId, resource, method, authResource, authByExperiment = true) => {
+  const isV2 = resource.startsWith('/v2/');
+
   let experimentId = authResource;
-  if (!authByExperiment) {
+
+  // authResource is always experimentId in V2
+  if (!authByExperiment && !isV2) {
     const experiments = await projectService.getExperiments(authResource);
     experimentId = experiments[0].experimentId;
   }
 
-  const granted = await accessService.canAccessExperiment(userId,
-    experimentId,
-    resource,
-    method);
+  let granted;
+
+  if (isV2) {
+    granted = await userAccess.canAccessExperiment(userId, experimentId);
```

Comment:
what about the more granular access permissions based on what resource and method the user wants to access?

Reply:
You're right, added role checking